### DISCUSSION TITLE
[Feature] Single file with separate attachments

### DISCUSSION THREAD

**Discussion by MindaugasMateika** [allure-framework/allure2#2227] [Ideas & Feedback]
### Describe the Bug

Single file option is pretty great for a smaller reports. However, if you have a thousand of reports with videos and screenshots then the single file is 1-2GB+ of size and becomes unusable in the browser due to lag or unresponsiveness.

I would like to propose an idea for another similar option: single file with separate attachments. Idea is based on how Playwright constructs html reports.

Proposed structure - index.html would have all information about test run and, if possible, base64 encoded text attachments. `data` folder would have all the heavier attachments - videos, screenshots, zips, etc. Index report would reference attachments like `src="data/attachment-123.png"` and so on.

📂 allure-report
--📂 data
----📄 attachment-abc.mp4
----📄 attachment-123.png
----📄 attachment-xyz.zip
----...
--📄 index.html

### Steps to Reproduce

-

### Expected Behaviour

-

### What Language are you using?

Kotlin

### What Framework/Allure Integration you are using?

allure-testng

### What version of Allure Integration you are using?

2.24.1

### What version of Allure Report you are using?

2.24.1

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by MindaugasMateika** [comment]
@baev do you have any tips and pointers on what files/behavior would need changing? I'd like to fork and try this approach. If you would be open for contribution, I could try make a PR later on.

**Reply by baev** [reply]
It's quite tricky at this point. 

First, you need to write attachments into the file system. The logic is located in [ReportWebGenerator](https://github.com/allure-framework/allure2/blob/master/allure-generator/src/main/java/io/qameta/allure/core/ReportWebGenerator.java) class. Still, the way it is currently implemented requires some additional API improvements to add the ability to choose what files should be stored on the file system and what should go into index.html.

Then, you need to solve a similar problem on the front-end side to determine whether you should look for the file contents in index.html or try to read them from the file system. See the [loader.](https://github.com/allure-framework/allure2/blob/master/allure-generator/src/main/javascript/data/loader.js)

It could probably be worked around by adding a fallback to the file system file if no file is detected in `index.html, ' but it may cause other issues.

**Comment by baev** [comment]
I like the proposition in general. 

But for the huge reports, I would always recommend having some other report storage, such as storing all the reports in an S3 bucket. In that case, you'll never need to use a single-file mode.

For thousands of reports, please use [Allure TestOps](https://qameta.io/).
